src/drivers/riscv_plic0.c

```diff
@@ -64,6 +67,19 @@ static __inline__ int __metal_plic0_disable(int hartid, int id) {
               (id >> METAL_PLIC_SOURCE_SHIFT) * 4) &=
         ~(1 << (id & METAL_PLIC_SOURCE_MASK));
 
+    /* Check if any PLIC interrupts remain enabled and simply exit
+     * if they are. */
+    for (int i = 0; i < METAL_RISCV_PLIC0_0_RISCV_NDEV % 32; i++) {
+        if (PLIC_REGW(METAL_RISCV_PLIC0_ENABLE_BASE +
```

Comment:
I'm confused about this -- you're looping over 'i', but not using 'i' in the loop body?